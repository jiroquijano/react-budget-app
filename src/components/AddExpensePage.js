import React from 'react';
import ExpenseForm from './ExpenseForm';
import {connect} from 'react-redux';
import {startAddExpense} from '../actions/expenses';

export class AddExpensePage extends React.Component{
    onSubmit = (expense) =>{
        this.props.addExpenseDispatch(expense);
        this.props.history.push("/");
    }

    render(){
        return (
            <div>
                <h1>Expense page</h1>
                <ExpenseForm onSubmit={this.onSubmit}/>
            </div>
        );
    };
}

// const AddExpensePage = (props) =>(
//     <div>
//         <h1>Expense page</h1>
//         <ExpenseForm onSubmit={(expense)=>{
//             //props.dispatch(addExpense(expense)); 
//             props.onSubmit(expense); //thanks to mapDispatchToProps from react-redux
//             props.history.push("/");
//         }}/>
//     </div>
// );

const mapDispatchToProps = (dispatch)=>({
        addExpenseDispatch: (expense) => dispatch(startAddExpense(expense))
});

const ConnectedAddExpensePage = connect(undefined,mapDispatchToProps)(AddExpensePage);

export default ConnectedAddExpensePage;