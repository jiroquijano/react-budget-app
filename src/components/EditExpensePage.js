import React from 'react';
import {connect} from 'react-redux';
import ExpenseForm from './ExpenseForm';
import {editExpense,startRemoveExpense} from '../actions/expenses';

export class EditExpensePage extends React.Component{

    onSubmit = (expense) => {
        this.props.editExpenseDispatch(this.props.expense.id, expense);
        this.props.history.push("/");
    }

    onClick = () =>{
        this.props.removeExpenseDispatch(this.props.expense.id);
        this.props.history.push("/");
    }

    render(){ 
        return (<div>
                    Edit Expense Page for id: {this.props.expense.id}
                    <ExpenseForm 
                        expense={this.props.expense}
                        onSubmit={this.onSubmit}
                    />
                    <button onClick={this.onClick}>Remove</button>
                </div>);
    };
};

const mapStateToProps = (state,props) =>({
    expense: state.expenses.find((expense)=>expense.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch) =>({
    editExpenseDispatch : (id, expense) => dispatch(editExpense(id,expense)),
    removeExpenseDispatch : (id) => dispatch(startRemoveExpense(id))
})

const ConnectedEditExpensePage = connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);

export default ConnectedEditExpensePage;