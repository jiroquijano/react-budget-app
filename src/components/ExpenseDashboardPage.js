import React from 'react';

const ExpenseDashboardPage = () =>(
    <div>
    This is from the jiro's dashboard component
    </div>
);

export default ExpenseDashboardPage;