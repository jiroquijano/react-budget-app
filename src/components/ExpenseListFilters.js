import React from 'react';
import {connect} from 'react-redux'
import {setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate} from '../actions/filters';
import {DateRangePicker} from 'react-dates';

export class ExpenseListFilters extends React.Component{
    state = {
        calendarFocused : null
    };

    onDatesChange = ({startDate, endDate}) =>{
        this.props.setStartDateDispatch(startDate);
        this.props.setEndDateDispatch(endDate);
    };

    onFocusChange = (calendarFocused) =>{
        this.setState(()=>({
            calendarFocused
        }));
    };

    dispatchSetTextFilter = (event) => {
        this.props.setTextFilterDispatch(event.target.value);
    }

    sortByFilter = (event) => {
        event.target.value === 'date' ? this.props.sortByDateDispatch() : this.props.sortByAmountDispatch();
    }

    render(){
        return(
            <div className="content-container">
                <div className="input-group">
                    <div className="input-group__item">
                        <input 
                            className="text-input"
                            type="text"
                            value={this.props.filters.text}
                            onChange={this.dispatchSetTextFilter}
                            placeholder="Search expenses"
                        />
                    </div>

                    <div className="input-group__item">
                        <select
                            className="select"
                            value={this.props.filters.sortBy}
                            onChange={this.sortByFilter}
                        >
                            <option value = "date">Date</option>
                            <option value = "amount">Amount</option>
                        </select>
                    </div>

                    <div className="input-group__item">
                        <DateRangePicker
                            startDate = {this.props.filters.startDate}
                            startDateId = "startdate"
                            endDate = {this.props.filters.endDate}
                            endDateId= "enddate"
                            onDatesChange = {this.onDatesChange}
                            focusedInput = {this.state.calendarFocused}
                            onFocusChange = {this.onFocusChange}
                            showClearDates = {true}
                            numberOfMonths = {1}
                            isOutsideRange = {()=>false}
                        />
                    </div>
                </div> 
            </div>
        );
    };
};

const mapStateToProps = (state) => ({
    filters: state.filters
});

const mapDispatchToProps = (dispatch) =>({
    setTextFilterDispatch : (filter)=> dispatch(setTextFilter(filter)),
    sortByDateDispatch : () => dispatch(sortByDate()),
    sortByAmountDispatch : () => dispatch(sortByAmount()),
    setStartDateDispatch : (startDate) => dispatch(setStartDate(startDate)),
    setEndDateDispatch : (endDate) => dispatch(setEndDate(endDate))
})

const ConnectedExpenseListFilters = connect(mapStateToProps,mapDispatchToProps)(ExpenseListFilters);

export default ConnectedExpenseListFilters;