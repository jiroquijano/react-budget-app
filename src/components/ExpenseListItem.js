import React from 'react';
import {connect} from 'react-redux';
import {removeExpense} from '../actions/expenses';

const ExpenseListItem = ({id, description, amount, createdAt,dispatch}) => (
    <div>
        <h3>{description}</h3>
        <p>Amount: {amount} | Created at: {createdAt}</p>
        <button onClick={()=>{
            dispatch(removeExpense(id));
        }}>Remove</button>
    </div>
);

const ConnectedExpenseListItem = connect()(ExpenseListItem);

export default ConnectedExpenseListItem;