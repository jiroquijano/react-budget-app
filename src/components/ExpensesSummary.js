import React from 'react';
import {connect} from 'react-redux';
import getVisibleExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';
import {Link} from 'react-router-dom';

export class ExpensesSummary extends React.Component{
    render(){
        return (
            <div className="page-header">
                <div className="content-container">
                    <h1 className="page-header__title">Viewing <span>{this.props.expenseCount}</span> {this.props.expenseCount > 1 ? 'expenses' : 'expense'} totalling <span>{numeral(this.props.expensesTotal/100).format('$0,0.00')}</span> </h1>
                    <div className="page-header__actions">
                        <Link className="button" to='/create'>Add Expense</Link>
                    </div>
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state) =>{
    const expenses = getVisibleExpenses(state);
    return {
        expenseCount : expenses.length,
        expensesTotal : getExpensesTotal(expenses)
    };
};

const ConnectedExpensesSummary = connect(mapStateToProps)(ExpensesSummary);
export default ConnectedExpensesSummary;