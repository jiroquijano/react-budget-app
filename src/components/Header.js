import React from 'react';
import {NavLink} from 'react-router-dom';
import {connect} from 'react-redux';
import {startLogout} from '../actions/auth';

export const Header = (props) => (
    <header>
        <h1>Expensify App</h1>
        <NavLink activeClassName="is-active" exact to="/dashboard">DASHBOARD | </NavLink>
        <NavLink activeClassName="is-active" exact to="/create"> ADD EXPENSE | </NavLink>
        <NavLink activeClassName="is-active" exact to="/help"> HELP | </NavLink>
        <button onClick={props.startLogout}> Log out </button>
    </header>
        
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => {dispatch(startLogout())}
});

const ConnectedHeader = connect(undefined, mapDispatchToProps)(Header);

export default ConnectedHeader;