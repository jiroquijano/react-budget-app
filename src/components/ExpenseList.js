import React from 'react';
import {connect} from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import getVisibleExpenses from '../selectors/expenses';

export const ExpenseList = (props) =>(
    <div>
        <h1>Expense List</h1>
        {props.expenses.map(expense=>{
            return (
                <ExpenseListItem
                    key={expense.id} 
                    {...expense}
                />
            );
        })}
    </div>
);

const mapStatetoProps = (state)=>{
    return {
        expenses: getVisibleExpenses(state)
    };
};

export default connect(mapStatetoProps)(ExpenseList);
