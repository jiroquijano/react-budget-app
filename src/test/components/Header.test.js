import React from 'react';
import {shallow} from 'enzyme';
import {Header} from '../../components/Header';
import toJSON from 'enzyme-to-json';

test ("should render Header correctly", ()=>{
   const wrapper = shallow(<Header startLogout={()=>{}} />);
   expect(toJSON(wrapper)).toMatchSnapshot();
});

test ("should call logout prop on click", ()=>{
   const onClickSpy = jest.fn();
   const wrapper = shallow(<Header startLogout={onClickSpy}/>);
   wrapper.find('button').simulate('click');
   expect(onClickSpy).toHaveBeenCalled();
});
