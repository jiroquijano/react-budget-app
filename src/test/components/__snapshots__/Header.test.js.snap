// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header>
  <h1>
    Expensify App
  </h1>
  <NavLink
    activeClassName="is-active"
    exact={true}
    to="/"
  >
    HOME | 
  </NavLink>
  <NavLink
    activeClassName="is-active"
    exact={true}
    to="/create"
  >
     ADD EXPENSE | 
  </NavLink>
  <NavLink
    activeClassName="is-active"
    exact={true}
    to="/help"
  >
     HELP | 
  </NavLink>
  <button
    onClick={[Function]}
  >
     Log out 
  </button>
</header>
`;
